crates/core/tedge_actors/src/builders.rs

```diff
@@ -0,0 +1,232 @@
+use crate::mpsc;
+use crate::ClientId;
+use crate::ConcurrentServiceMessageBox;
+use crate::DynSender;
+use crate::KeyedSender;
+use crate::Message;
+use crate::NullSender;
+use crate::RequestResponseHandler;
+use crate::RuntimeError;
+use crate::RuntimeHandle;
+use crate::Sender;
+use crate::SenderVec;
+use crate::ServiceMessageBox;
+use crate::SimpleMessageBox;
+use async_trait::async_trait;
+use std::convert::Infallible;
+
+/// Materialize an actor instance under construction
+///
+/// Such an instance is:
+/// 1. built from some actor configuration
+/// 2. connected to other peers
+/// 3. eventually spawned into an actor.
+#[async_trait]
+pub trait ActorBuilder {
+    /// Build and spawn the actor
+    async fn spawn(self, runtime: &mut RuntimeHandle) -> Result<(), RuntimeError>;
+}
+
+/// Builder of `T`
+pub trait Builder<T>: Sized {
+    type Error: std::error::Error;
+
+    /// Builds the entity or returns an error
+    fn try_build(self) -> Result<T, Self::Error>;
+
+    /// Builds the entity or panics
+    fn build(self) -> T {
+        self.try_build().unwrap()
+    }
+}
+
+/// A trait to connect a message box under-construction to peer messages boxes
+pub trait MessageBoxConnector<Request: Message, Response: Message, Config: Default> {
```

Comment:
Sure, we can rename them.

I came with these names because they helped me not swapping by mistake input for output in the method types. Indeed the input of the service is the output of the client, while the name request is meaningful for both.